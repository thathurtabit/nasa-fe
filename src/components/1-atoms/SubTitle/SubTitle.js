import React from 'react';
import PropTypes from 'prop-types';
import Title from './SubTitle.styled';

const SubTitle = ({ title }) => <Title>{title}</Title>;

export default SubTitle;

SubTitle.propTypes = {
  title: PropTypes.string.isRequired,
};
