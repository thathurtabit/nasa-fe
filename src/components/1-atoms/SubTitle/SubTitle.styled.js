import styled from 'styled-components';

const Title = styled.h1`
  letter-spacing: 2px;
  font-size: calc(${props => props.theme.fonts.baseSize} + 0.5vmin);
  position: relative;
  text-transform: uppercase;
`;

export default Title;
