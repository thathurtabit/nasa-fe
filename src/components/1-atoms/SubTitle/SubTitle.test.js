import React from 'react';
import { shallowToJson } from 'enzyme-to-json';
import 'jest-styled-components';
import {
  shallowWithTheme,
  mountWithTheme,
} from '../../../utils/test_config/testHelpers';

import SubTitle from './SubTitle';

const Title = 'Test Subtitle';

describe('SubTitle', () => {
  it('should render correctly', () => {
    const element = shallowWithTheme(<SubTitle title={Title} />);
    expect(shallowToJson(element)).toMatchSnapshot();
  });

  it('should return the correct text', () => {
    const element = mountWithTheme(<SubTitle title={Title} />);
    expect(element.text()).toBe(Title);
  });
});
