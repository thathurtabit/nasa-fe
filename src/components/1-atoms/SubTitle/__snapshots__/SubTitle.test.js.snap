// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubTitle should render correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
